service/history/visibilityQueueTaskExecutor.go

```diff
@@ -436,9 +439,34 @@ func (t *visibilityQueueTaskExecutor) processDeleteExecution(
 		StartTime:   task.StartTime,
 		CloseTime:   task.CloseTime,
 	}
+	if t.shard.GetConfig().VisibilityProcessorEnsureCloseBeforeDelete() {
+		if !t.isVisibilityClosed(task) {
+			return consts.ErrTaskRetry
```

Comment:
Need to return the error defined in https://github.com/temporalio/temporal/pull/3405

that's why I believe this and 3405 should be merged into one PR.

Reply:
Merged that branch into here and changed the error. I might land the other one first, though, since it's already approved, and then I'll just rebase.